### ISSUE TITLE
Question: Comprehensiveness of Web Accessibility Checklist

### ISSUE THREAD

**Issue by nas5w** [a11yproject/a11yproject.com#736]
Hi all, fantastic work on the a11yproject.com site and I look forward to the redesign! 

I was browsing the [Web Accessibility Checklist](https://a11yproject.com/checklist) and noted that, while everything on it seems really sound and in accordance with WCAG guidance, there are potentially missing items. For example, one item I would expect to see is in reference to ficker/flash/blink rate of any element. In the past, I have used checklists like the [webaim.org checklist](https://webaim.org/standards/wcag/checklist).

Of course, presenting a checklist with 100 items on it could easily scare off a newcomer. So my question is: how comprehensive is the [Web Accessibility Checklist](https://a11yproject.com/checklist) intended to be? Are you open to adding more items? Some of the more important items that I think might be missing involve:

- [Content linearization](https://webaim.org/techniques/screenreader/#linearization). You already have this in the forms section but really it applies to the entire application.
- [Instructions shouldn't depend on visual or audio](https://www.w3.org/TR/WCAG21/#sensory-characteristics). E.g., "click on the square to continue."
- [Color is not the only visual means of conveying information](https://www.w3.org/TR/WCAG21/#use-of-color). E.g., Red shouldn't mean "danger" without some accompanying text.
- [Pause Stop Hide](https://www.w3.org/TR/WCAG21/#pause-stop-hide) and [Three flashes or below](https://www.w3.org/TR/WCAG21/#three-flashes-or-below-threshold)
- [Images of text](https://www.w3.org/TR/WCAG21/#images-of-text). If text can convey the text in an image, consider just using the text.
- [No parsing errors](https://www.w3.org/TR/WCAG21/#parsing)

Thank you!

**Comment by scottaohara** [comment]
Hey @nas5w, thanks for opening the issue!

The checklist has actually been a topic we've started talking about, and definitely will be getting some revising with the current website redesign project.  So this is definitely a good time to be raising ideas for improving it.

cc @svinkle @ericwbailey

**Comment by nas5w** [comment]
Great, thank you! The WCAG 2.1 guideline is huge, so it's hard to envision a comprehensive checklist that isn't also huge. One ambitious idea I've had is to have a checklist generator of sorts: answer questions about your application and a checklist with only applicable items is generated. Not sure how that would work in practice!

**Comment by sebacicltd** [comment]
I'm new to A11y, and I'm coming to it from a language perspective, but I'm
really keen on a checklist too, as I'll be building websites for local
government services.

I love the idea of the questionnaire to narrow down the need to check every
item (Any flashing? NO. Ignore Guideline 2.3 Seizures and Physical
Reactions)

Checking through this guideline, I've just learnt that parallax scrolling
is off the menu for my sites. But all the modern DIY website builders love
this "feature".

We really DO need to make the guideline "accessible" to people new to A11y.

regards

Jonathan Holden

**Comment by ericwbailey** [comment]
Thank you for the feedback, @sebacicltd. That is definitely one of the bigger-picture goals of the redesign, one that we've been actively discussing. 

The problem to beat is the unknown unknowns, as you illustrated with parallax scrolling. And as @nas5w pointed out WCAG is comprehensive to the point of being difficult to get a quick answer, too.

We want to make the checklist good enough that it can be run through without causing someone to abandon the task on account of complexity, yet thorough enough that it covers a good amount of accessibility criteria. It's a tough line to walk, so I'll definitely be tapping both of you for reviews and input as we update, if you're willing.

**Comment by nas5w** [comment]
Yes please! I would love to be involved in whatever way could help.

**Comment by eschafer** [comment]
I like how the [WCAG Quick Reference](https://www.w3.org/WAI/WCAG21/quickref/?currentsidebar=%23col_customize) includes filters to narrow things down a bit. The way it's presented is still too much info at once for me, but I'd love if it was somehow simplified so that I could get a better overview of what's most relevant to what I'm working on.

I just realized that the A11y Project's checklist is similar to an Accessibility Insights assessment, right? I'm just thinking about how I haven't used the A11y Project's checklist because I know it's incomplete (Accessibility Insights is supposed to cover everything in WCAG AA). It doesn't seem worth it to go through the whole thing and not know what I'm still missing at the end.

**Comment by ericwbailey** [comment]
I don't believe it covers the full WCAG AA, but it is far more comprehensive in its current state than it was before. I'm unfamiliar with the Accessibility Insights checklist tool, but there typically is a lot of overlap with a lot of checklist-style tools as they try to address common inaccessible code and design patterns.

I would argue that any amount of checklist work on either tool would leave you in a better place than doing nothing at all.

**Comment by ericwbailey** [comment]
Closing for inactivity.
